actors/builtin/miner/miner_test.go

```diff
@@ -1319,153 +1311,250 @@ func TestDeclareFaults(t *testing.T) {
 		// Get sector into proving state
 		rt := builder.Build(t)
 		actor.constructAndVerify(rt)
-		precommits := actor.commitAndProveSectors(rt, 1, 100, nil)
+		allSectors := actor.commitAndProveSectors(rt, 1, 181, nil)
+		pwr := miner.PowerForSectors(actor.sectorSize, allSectors)
 
-		// Skip to end of proving period, cron adds sectors to proving set.
-		completeProvingPeriod(rt, actor, &cronConfig{})
-		info := actor.getSector(rt, precommits[0].SectorNumber)
+		// add lots of funds so penalties come from vesting funds
+		initialLocked := big.Mul(big.NewInt(200), big.NewInt(1e18))
+		actor.addLockedFunds(rt, initialLocked)
+
+		// find deadline for sector
+		st := getState(rt)
+		dlIdx, _, err := st.FindSector(rt.AdtStore(), allSectors[0].SectorNumber)
+		require.NoError(t, err)
+
+		// advance to first proving period and submit so we'll have time to declare the fault next cycle
+		advanceAndSubmitPoSts(rt, actor, allSectors...)
 
 		// Declare the sector as faulted
-		actor.declareFaults(rt, info)
+		actor.declareFaults(rt, allSectors...)
 
-		// TODO: advance to window post and observe fee
-		//
-		//ss, err := info.SealProof.SectorSize()
-		//require.NoError(t, err)
-		//sectorQAPower := miner.QAPowerForSector(ss, info)
-		//fee := miner.PledgePenaltyForDeclaredFault(actor.epochRewardSmooth, actor.epochQAPowerSmooth, sectorQAPower)
+		// faults are recorded in state
+		dl := actor.getDeadline(rt, dlIdx)
+		assert.True(t, pwr.Equals(dl.FaultyPower))
+
+		// Skip to end of proving period.
+		dlinfo := actor.deadline(rt)
+		for dlinfo.Index != dlIdx {
+			dlinfo = advanceDeadline(rt, actor, &cronConfig{})
+		}
```

Comment:
it would be nice to have an `actor.advanceToDeadline(dlIdx)` helper.

Reply:
I tried to do this, but every way I tried to do it that covered all the cases ended up being more total lines of code and less clarity.